api/api.go

```diff
@@ -285,6 +285,58 @@ func (c *Client) SetRegion(region string) {
 	c.config.Region = region
 }
 
+// GetNodeClient returns a new Client that will dial the specified node. If the
+// QueryOptions is set, the function will ensure that it is initialized and
+// that the Params field is valid.
+func (c *Client) GetNodeClient(nodeID string, q **QueryOptions) (*Client, error) {
+	node, _, err := c.Nodes().Info(nodeID, &QueryOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if node.Status == "down" {
+		return nil, NodeDownErr
+	}
+	if node.HTTPAddr == "" {
+		return nil, fmt.Errorf("http addr of node %q (%s) is not advertised", node.Name, nodeID)
+	}
+
+	region := ""
+	if q != nil && *q != nil && (*q).Region != "" {
+		region = (*q).Region
+	} else if c.config.Region != "" {
+		// Use the region from the client
+		region = c.config.Region
+	} else {
+		// Use the region from the agent
+		agentRegion, err := c.Agent().Region()
```

Comment:
This seems odd. There may not be an agent if this is being embedded in another program and it would cause an error. It should just do nothing as the http handler would default to the global region

Reply:
Won't this query the agent at NOMAD_ADDR for the region? That seems potentially useful even in embedded circumstances.

(This logic was cargo culted from the previous location, so I may not fully grok it)